stdlib/src/builtin/int_literal.mojo

```diff
@@ -269,6 +269,47 @@ struct IntLiteral(
         """
         return self
 
+    @always_inline("nodebug")
+    fn __divmod__(self, rhs: Self) -> Tuple[Self, Self]:
+        """Return the quotient and remainder of the division of self by rhs.
+
+        Args:
+            rhs: The value to divide on.
+
+        Returns:
+            The quotient and remainder of the division.
+        """
+        var quotient: Self = self.__floordiv__(rhs)
+        var remainder: Self = self - (quotient * rhs)
+        return quotient, remainder
+
+    @always_inline("nodebug")
+    fn __round__(self, ndigits: Int) -> Self:
+        """Return the rounded value of the IntLiteral value, which is itself.
+
+        Args:
+            ndigits: The number of digits to round to.
+
+        Returns:
+            The IntLiteral value itself if ndigits >= 0 else the rounded value.
+        """
+        if ndigits >= 0:
+            return self
+        alias one = __mlir_attr.`#kgen.int_literal<1> : !kgen.int_literal`
+        alias ten = __mlir_attr.`#kgen.int_literal<10> : !kgen.int_literal`
+        var multiplier = one
+        # TODO: Use IntLiteral.__pow__() when it's implemented.
+        for _ in range(-ndigits):
+            multiplier = __mlir_op.`kgen.int_literal.binop`[
+                oper = __mlir_attr.`#kgen<int_literal.binop_kind mul>`
+            ](multiplier, ten)
+        alias Pair = Tuple[Self, Self]
+        var t: Pair = self.__divmod__(Self(multiplier))
+        if t[1] * 2 >= multiplier:
+            t[0] += 1
+            t[1] -= multiplier
+        return self - t[1]
```

Comment:
I'm running into compiler crashes with this internally, and I'm not even sure how it could pass CI. It probably has to do with `Tuple` and `IntLiteral`.

Anyway, it seems you don't actually use t[0], right? Could you please remove `__divmod__` and change this to just use `%`?

Reply:
Sure, It's probably because a problem I had in the beginning and now something was implemented in upstream so now it could be simplified:

Done: https://github.com/msaelices/mojo/commit/e6f55f290b4c1baef25b17ef11fd3d6053d088e9